modules/clientutils.js

```diff
@@ -659,6 +644,49 @@
         };
 
         /**
+         * Makes selector by defined type XPath, Name or Label. Function has same result as selectXPath in Casper module for 
+         * XPath type - it makes XPath object. 
+         * Function also accepts name attribut of the form filed or can select element by its label text.
+         *
+         * @param  String selector Selector of defined type
+         * @param  String|null  type Type of selector, it can have these values:     
+         *         css - CSS3 selector - selector is returned trasparently
+         *         xpath - XPath selector - return XPath object    
+         *         name|names - select input of specific name, internally covert to CSS3 selector
+         *         label|labels - select input of specific label, internally covert to XPath selector. As selector is label's text used.
+         * @return String|Object          
+         */
+        this.makeSelector = function makeSelector(selector, type){
+            type = type || 'xpath'; // default type
+            var ret;
+
+            if (typeof selector === "object") { // selector object (CSS3 | XPath) could by passed
+                selector = selector.path;
+            }
+
+            switch (type) {
+                case 'css': // do nothing
+                    ret = selector;
+                    break;
+                case 'name': // convert to css 
+                case 'names': 
+                    ret = '[name="' + selector + '"]';
```

Comment:
The reason was that fill() did a lot of things and I needed similar functionality for my methods - that's why I isolated of making selector (makeSelector method). 
"Finders" needed more parameters, in all cases is findAll() called etc. On other hand makeSelector() is simple method with clear interface.  
I think using switch in this case is OK but I'll would rewrite it if you want.
(btw. I think in method setField() is bad example of using mega switch with a lot of logic)